### DISCUSSION TITLE
timeout for produce action

### DISCUSSION THREAD

**Discussion by mcat345** [fastlane/fastlane#18898] [Q&A]
is there a way to set a timeout for action if it's running too long?

**Comment by crazymanish** [comment]
Hi @mcat345 

The awesome part of Fastlane is, you can use any ruby code...for time, just use the `timeout` gem https://github.com/ruby/timeout

let say, you want to timeout `produce` action after 10seconds 

```ruby
lane :produce_with_timeout do
  require 'timeout'
  Timeout::timeout(10) do
    produce # this can be any fastlane action
  end
end
